cdap-tms/src/main/java/co/cask/cdap/messaging/service/CoreMessagingService.java

```diff
@@ -221,22 +221,31 @@ protected void startUp() throws Exception {
     Queue<TopicId> asyncCreationTopics = new LinkedList<>();
 
     for (String topic : new HashSet<>(cConf.getTrimmedStringCollection(Constants.MessagingSystem.SYSTEM_TOPICS))) {
-      TopicId topicId;
-      try {
-        topicId = NamespaceId.SYSTEM.topic(topic);
-      } catch (IllegalArgumentException e) {
-        // Ignore invalid topic
-        LOG.warn("Ignore creation of invalid topic '{}'.", topic);
-        continue;
-      }
-
-      try {
-        createTopicIfNotExists(topicId);
-      } catch (Exception e) {
-        // If failed, add it to a list so that the retry will happen asynchronously
-        LOG.warn("Topic {} creation failed with exception {}. Will retry.", topicId, e.getMessage());
-        LOG.debug("Topic {} creation failure stacktrace", topicId, e);
-        asyncCreationTopics.add(topicId);
+      // Create topics with topic prefix and top partition
+      if (topic.contains(":")) {
+        String[] topicPrefixPartition = topic.split(":");
+        Integer partition;
+        if (topicPrefixPartition.length != 2) {
+          LOG.warn("Ignore creation of invalid topic '{}'.", topic);
+          continue;
+        }
+        String topicPrefix = topicPrefixPartition[0];
+        try {
+          partition = Integer.valueOf(topicPrefixPartition[1]);
+          if (partition < 0) {
+            throw new IllegalArgumentException();
+          }
+          NamespaceId.SYSTEM.topic(topicPrefix);
+        } catch (IllegalArgumentException e) {
```

Comment:
to clarify, you shouldn't throw a NumberFormatException yourself. That is just what parseInt() and valueOf() throw. The diff has gotten all messed up now after the rebase so I'm having difficulty remembering what it was like when I commented. At the end of the day, it should log a descriptive error message that tells the user exactly what is wrong. If the partitions is 0 or less, it should say something about partitions needing to be positive. If it can't parse it, it should say the partitions could not be parsed as an integer.